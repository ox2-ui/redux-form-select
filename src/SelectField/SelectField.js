import React, { PropTypes } from 'react';
import { default as Select } from '@ox2/select/SelectField';
import WrapperLabel from '@ox2/ui/WrapperLabel';
/**
 * SelectField Component
 */
const SelectField = ({ input, label, placeholder, ...custom }) => {
  return (
    <WrapperLabel text={label || ''}>
      <Select
        placeholder={placeholder || ''}
        value={input.value}
        onUpdate={(value) => input.onChange(value)}
        {...custom}
      />
    </WrapperLabel>
  );
};

SelectField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
};

export default SelectField;
