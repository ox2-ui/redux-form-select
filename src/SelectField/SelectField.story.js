import React from 'react';
import { storiesOf } from '@kadira/storybook';
import SelectField from './SelectField';

const Groups = [
  { label: 'Speakers', value: 'speakers' },
  { label: 'Attendies', value: 'attendies' },
  { label: 'Moderators', value: 'moderators' },
];

const input = {
  value: 'speakers',
  onChange: (value) => console.log('🍇', value), // eslint-disable-line no-console
};

storiesOf('SelectField', module)
  .add('single', () => (
    <SelectField input={input} options={Groups} />
  ))
  .add('multi', () => (
    <SelectField
      input={input}
      options={Groups}
      multi={true}
      simpleValue={true}
    />
  ));
