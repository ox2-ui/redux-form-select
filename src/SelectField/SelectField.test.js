/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import SelectField from './SelectField';

const Groups = [
  { label: 'Speakers', value: 'speakers' },
  { label: 'Attendies', value: 'attendies' },
  { label: 'Moderators', value: 'moderators' },
];

const input = {
  value: 'speakers',
  onChange: (value) => console.log('🍇', value), // eslint-disable-line no-console
};

it('renders correctly', () => {
  const wrapper = mount(
    <SelectField
      input={input}
      options={Groups}
      placeholder={'Select groups'}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
