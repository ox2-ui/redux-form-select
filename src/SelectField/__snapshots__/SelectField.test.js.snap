exports[`test renders correctly 1`] = `
<SelectField
  input={
    Object {
      "onChange": [Function],
      "value": "speakers",
    }
  }
  options={
    Array [
      Object {
        "label": "Speakers",
        "value": "speakers",
      },
      Object {
        "label": "Attendies",
        "value": "attendies",
      },
      Object {
        "label": "Moderators",
        "value": "moderators",
      },
    ]
  }
  placeholder="Select groups">
  <WrapperLabel
    text="">
    <styled.div>
      <div
        className="hgoCNT">
        <styled.div>
          <div
            className="eFyCaC" />
        </styled.div>
        <Styled(Select)
          onUpdate={[Function]}
          options={
            Array [
              Object {
                "label": "Speakers",
                "value": "speakers",
              },
              Object {
                "label": "Attendies",
                "value": "attendies",
              },
              Object {
                "label": "Moderators",
                "value": "moderators",
              },
            ]
          }
          placeholder="Select groups"
          value="speakers">
          <Select
            className="boQsxg"
            onUpdate={[Function]}
            options={
              Array [
                Object {
                  "label": "Speakers",
                  "value": "speakers",
                },
                Object {
                  "label": "Attendies",
                  "value": "attendies",
                },
                Object {
                  "label": "Moderators",
                  "value": "moderators",
                },
              ]
            }
            placeholder="Select groups"
            value="speakers">
            <Styled(Select)
              className="boQsxg"
              onChange={[Function]}
              onUpdate={[Function]}
              optionRenderer={[Function]}
              options={
                Array [
                  Object {
                    "label": "Speakers",
                    "value": "speakers",
                  },
                  Object {
                    "label": "Attendies",
                    "value": "attendies",
                  },
                  Object {
                    "label": "Moderators",
                    "value": "moderators",
                  },
                ]
              }
              placeholder="Select groups"
              value="speakers"
              valueRenderer={[Function]}>
              <Select
                addLabelText="Add \"{label}\"?"
                arrowRenderer={[Function]}
                autosize={true}
                backspaceRemoves={true}
                backspaceToRemoveMessage="Press backspace to remove {label}"
                className="boQsxg fUPKPg"
                clearAllText="Clear all"
                clearRenderer={[Function]}
                clearValueText="Clear value"
                clearable={true}
                deleteRemoves={true}
                delimiter=","
                disabled={false}
                escapeClearsValue={true}
                filterOptions={[Function]}
                ignoreAccents={true}
                ignoreCase={true}
                inputProps={Object {}}
                isLoading={false}
                joinValues={false}
                labelKey="label"
                matchPos="any"
                matchProp="any"
                menuBuffer={0}
                menuRenderer={[Function]}
                multi={false}
                noResultsText="No results found"
                onBlurResetsInput={true}
                onChange={[Function]}
                onCloseResetsInput={true}
                onUpdate={[Function]}
                openAfterFocus={false}
                optionComponent={[Function]}
                optionRenderer={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Speakers",
                      "value": "speakers",
                    },
                    Object {
                      "label": "Attendies",
                      "value": "attendies",
                    },
                    Object {
                      "label": "Moderators",
                      "value": "moderators",
                    },
                  ]
                }
                pageSize={5}
                placeholder="Select groups"
                required={false}
                scrollMenuIntoView={true}
                searchable={true}
                simpleValue={false}
                tabSelectsValue={true}
                value="speakers"
                valueComponent={[Function]}
                valueKey="value"
                valueRenderer={[Function]}>
                <div
                  className="Select boQsxg fUPKPg Select--single is-searchable has-value">
                  <div
                    className="Select-control"
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}>
                    <span
                      className="Select-multi-value-wrapper"
                      id="react-select-2--value">
                      <Value
                        disabled={false}
                        id="react-select-2--value-item"
                        instancePrefix="react-select-2-"
                        onClick={null}
                        value={
                          Object {
                            "label": "Speakers",
                            "value": "speakers",
                          }
                        }>
                        <div
                          className="Select-value">
                          <span
                            aria-selected="true"
                            className="Select-value-label"
                            id="react-select-2--value-item"
                            role="option">
                            <div
                              style={
                                Object {
                                  "alignItems": "center",
                                  "display": "inline-flex",
                                  "fontFamily": "Roboto Condensed",
                                }
                              }>
                              <div
                                style={
                                  Object {
                                    "fontSize": "16px",
                                  }
                                }>
                                Speakers
                              </div>
                            </div>
                          </span>
                        </div>
                      </Value>
                      <AutosizeInput
                        aria-activedescendant="react-select-2--value"
                        aria-expanded="false"
                        aria-haspopup="false"
                        aria-owns=""
                        className="Select-input"
                        minWidth="5"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        role="combobox"
                        value="">
                        <div
                          className="Select-input"
                          style={
                            Object {
                              "display": "inline-block",
                            }
                          }>
                          <input
                            aria-activedescendant="react-select-2--value"
                            aria-expanded="false"
                            aria-haspopup="false"
                            aria-owns=""
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                            role="combobox"
                            style={
                              Object {
                                "boxSizing": "content-box",
                                "width": "5px",
                              }
                            }
                            value="" />
                          <div
                            style={
                              Object {
                                "height": 0,
                                "left": 0,
                                "overflow": "scroll",
                                "position": "absolute",
                                "top": 0,
                                "visibility": "hidden",
                                "whiteSpace": "pre",
                              }
                            } />
                        </div>
                      </AutosizeInput>
                    </span>
                    <span
                      aria-label="Clear value"
                      className="Select-clear-zone"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      title="Clear value">
                      <span
                        className="Select-clear"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&times;",
                          }
                        } />
                    </span>
                    <span
                      className="Select-arrow-zone"
                      onMouseDown={[Function]}>
                      <span
                        className="Select-arrow"
                        onMouseDown={[Function]} />
                    </span>
                  </div>
                </div>
              </Select>
            </Styled(Select)>
          </Select>
        </Styled(Select)>
      </div>
    </styled.div>
  </WrapperLabel>
</SelectField>
`;
